### PR TITLE
fix(coordinator): add compatible logic to jwt when curie prover not adding public_key in login request

### PR DESCRIPTION
**Pull request by amoylan2**
### Purpose or design rationale of this PR

add compatible logic to jwt when curie prover not adding public_key in login request

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [x] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [ ] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: coordinator/internal/controller/api/auth.go [hunk 1]
```diff
@@ -58,8 +59,19 @@ func (a *AuthController) PayloadFunc(data interface{}) jwt.MapClaims {
...
 	}
 
+	publicKey := v.PublicKey
+	if publicKey == "" {
+		var err error
+		publicKey, err = v.RecoverPublicKeyFromSignature()
+		if err != nil {
+			// do not handle error here since already called v.Verify() beforehands so there should be no error
+			// add log just in case some error happens
+			log.Error("failed to recover public key from signature", "error", err.Error())
```
Comment (colinlyguo) [inline coordinator/internal/controller/api/auth.go]: we can still return error and in the log prints it's an impossible path.

**Review by colinlyguo — DISMISSED** [review]

File: coordinator/internal/controller/api/auth.go (hunk 1 above)
Comment (georgehao) [inline coordinator/internal/controller/api/auth.go:69]: I think here need return error

Comment (amoylan2) [inline coordinator/internal/controller/api/auth.go]: this method is part of jwt components and no error can be returned in response signature...

Comment (amoylan2) [reply coordinator/internal/controller/api/auth.go]: this method is part of jwt components and no error can be returned in response signature...

**Review by georgehao — DISMISSED** [review]

**Review by georgehao — DISMISSED** [review]

**Review by georgehao — APPROVED** [review]

**Review by colinlyguo — APPROVED** [review]

**Pull request merged**